src/rdb.c

```diff
@@ -2389,7 +2403,7 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, redisDb* db, int rdbflags,
         /* check for empty key (if all fields were expired) */
         if (hashTypeLength(o, 0) == 0) {
             decrRefCount(o);
-            goto emptykey;
+            goto expiredHash;
```

Comment:
hash length of 1 is not just due to expiry, it can also caused by some buggy commands or corrupted data, i think we should treat them differently.

Reply:
fixed. thanks.